packages/compass-query-bar/src/components/query-bar.spec.tsx

```diff
@@ -18,6 +35,61 @@ const exportToLanguageButtonId = 'query-bar-open-export-to-language-button';
 const queryHistoryButtonId = 'query-history-button';
 const queryHistoryComponentTestId = 'query-history';
 
+const getFilterInputEditorView = () => {
+  const filterInput = screen.getByTestId(
+    'query-bar-option-filter-input'
+  ) as any;
+  return filterInput._cm as EditorView;
+};
+
+const initiateFilterBarWithText = async (text: string) => {
+  const editorView = getFilterInputEditorView();
+  await waitUntilEditorIsReady(editorView);
+
+  editorView.dispatch({
+    changes: { from: 0, to: editorView.state.doc.length, insert: text },
+  });
+
+  await waitUntilEditorIsReady(editorView);
+};
+
+const getFilterInputContent = async () => {
+  const editorView = getFilterInputEditorView();
+  await waitUntilEditorIsReady(editorView);
+
+  return editorView.state.doc.sliceString(0) ?? '';
+};
+
+const getFilterInputEventHandler = () => {
+  const editorView = getFilterInputEditorView() as any;
+  return editorView.docView.dom as HTMLElement;
+};
+
+const clickOnFilterInputContent = () => {
+  userEvent.click(getFilterInputEventHandler());
+};
+
+async function eventuallyExpectFilterEditorToContain(
+  text: string,
+  timeout = 5,
+  times = 10
+) {
+  let result = '';
+
+  for (let i = 0; i < times; i++) {
+    await waitUntilEditorIsReady(getFilterInputEditorView());
+
+    result = await getFilterInputContent();
+    if (result.includes(text)) {
+      break;
+    }
+
+    await wait(timeout);
+  }
```

Comment:
I don't think you should rely on editor internals like that in test, testing-library comes with the `waitFor` method that you can use instead